modules/claims_api/lib/claims_api/error/soap_error_handler.rb

```diff
@@ -5,79 +5,74 @@
       return {} if @fault_string.include?('IntentToFileWebService') && @fault_string.include?('not found')
 
-      @reason = ''
-      unless person_found? == 'not found' || claim_found? == {} || processable? == 'unprocessable'
-        ClaimsApi::Logger.log("Returning 500 via local_bgs & soap_error_handler, fault_string:#{@fault_string},
-          with message: #{@fault_message}.")
-
-      end
-      @reason
+      get_exception
     end
 
-    def get_exception(fault_description)
-      case fault_description
-      when 'not_found'
+    private
+
+    def get_exception
+      if claim_not_found?
         raise ::Common::Exceptions::ResourceNotFound.new(detail: 'Resource not found.')
-      when 'unprocessable'
+      elsif bnft_claim_not_found?
+        {}
+      elsif unprocessable?
         raise ::Common::Exceptions::UnprocessableEntity.new(
           detail: 'Please try again after checking your input values.'
         )
-      when {}
-        {}
       else
+        ClaimsApi::Logger.log('soap_error_handler',
+                              detail: 'Returning 500 via local_bgs & soap_error_handler, ' \
+                                      "fault_string: #{@fault_string}, with message: #{@fault_message}, " \
+                                      "and fault_code: #{@fault_code}.")
         raise ::Common::Exceptions::ServiceError.new(detail: 'An external server is experiencing difficulty.')
       end
     end
 
-    private
-
-    def claim_found?
-      if @fault_string.include?('No BnftClaim found') ||
-         @fault_string.include?('not found')
-        @reason = {}
-      end
+    def claim_not_found?
+      errors = ['bnftClaimId-->bnftClaimId/text()', 'not found', 'No Person found']
+      has_errors = errors.any? { |error| @fault_string.include? error }
+      ClaimsApi::Logger.log('soap_error_handler',
+                            detail: 'Returning 404 via local_bgs & soap_error_handler, ' \
+                                    "fault_string: #{@fault_string}, with message: #{@fault_message}, " \
+                                    "and fault_code: #{@fault_code}.")
+      has_errors
     end
 
-    def person_found?
-      if @fault_string.include?('could not be converted') ||
-         @fault_string.include?('No Person found')
-        ClaimsApi::Logger.log("Returning 404 via local_bgs & soap_error_handler, fault_string:#{@fault_string},
-          with message: #{@fault_message}.")
-        @reason = 'not_found'
-      end
+    def bnft_claim_not_found?
+      errors = ['No BnftClaim found']
+      has_errors = errors.any? { |error| @fault_string.include? error }
+      ClaimsApi::Logger.log('soap_error_handler',
+                            detail: 'Returning 404 via local_bgs & soap_error_handler, ' \
+                                    "fault_string: #{@fault_string}, with message: #{@fault_message}, " \
+                                    "and fault_code: #{@fault_code}.")
```

Comment:
if has_errors?

Reply:
ya I was thinking it looked like it wanted something like that, I assume at least.  Since if there is no error I assume we wouldn't want any log out.